gapis/api/vulkan/destroy_resources_eos.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/memory"
+)
+
+type destroyResourcesAtEOS2 struct {
+}
+
+// NewdestroyResourcesAtEOS2 creates a transform that destroys all active
+// resources at the end of stream.
+func NewDestroyResourcesAtEOS2() *destroyResourcesAtEOS2 {
+	return &destroyResourcesAtEOS2{}
+}
+
+func (dropTransform *destroyResourcesAtEOS2) RequiresAccurateState() bool {
+	return false
+}
+
+func (dropTransform *destroyResourcesAtEOS2) BeginTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func (dropTransform *destroyResourcesAtEOS2) EndTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return append(inputCommands, dropTransform.addCleanUpCommands(ctx, inputState)...), nil
```

Comment:
Seems like having a separate method here increases the complexity.  Having a one-line that calls another function just to unpack the slice that it returns, feels like we could just inline the actual implementation. (Since as far as I can tell no-one else is calling it).